app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/hplus/HPlusHandlerThread.java

```diff
@@ -0,0 +1,446 @@
+package nodomain.freeyourgadget.gadgetbridge.service.devices.hplus;
+
+/*
+* @author João Paulo Barraca &lt;jpbarraca@gmail.com&gt;
+*/
+
+
+import android.content.Context;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.List;
+
+import nodomain.freeyourgadget.gadgetbridge.GBApplication;
+import nodomain.freeyourgadget.gadgetbridge.GBException;
+import nodomain.freeyourgadget.gadgetbridge.database.DBHandler;
+import nodomain.freeyourgadget.gadgetbridge.database.DBHelper;
+import nodomain.freeyourgadget.gadgetbridge.devices.hplus.HPlusConstants;
+import nodomain.freeyourgadget.gadgetbridge.devices.hplus.HPlusHealthSampleProvider;
+import nodomain.freeyourgadget.gadgetbridge.entities.DaoSession;
+import nodomain.freeyourgadget.gadgetbridge.entities.HPlusHealthActivityOverlay;
+import nodomain.freeyourgadget.gadgetbridge.entities.HPlusHealthActivityOverlayDao;
+import nodomain.freeyourgadget.gadgetbridge.entities.HPlusHealthActivitySample;
+import nodomain.freeyourgadget.gadgetbridge.impl.GBDevice;
+import nodomain.freeyourgadget.gadgetbridge.model.ActivityKind;
+import nodomain.freeyourgadget.gadgetbridge.model.ActivitySample;
+import nodomain.freeyourgadget.gadgetbridge.service.btle.TransactionBuilder;
+import nodomain.freeyourgadget.gadgetbridge.service.serial.GBDeviceIoThread;
+
+
+class HPlusHandlerThread extends GBDeviceIoThread {
```

Comment:
Actually this is far from perfect as events will be randomly skewed, and exceptions can break everything. Similar to what happens with a Timer.

This single thread will take care of some operations that should be done periodically, and with variable periodicity.
- Send a periodic keep alive packet ~60s
- Fetch Current Day data. This a burst of synchronous messages (1 get, 6 answers) plus a request/notification every 10 minutes
- Fetch Day Summaries ~ once per day
- Fetch Sleep Statistics ~ twice per day

I looked for an alternative way but this was the simplest I found, and saw no support for scheduling of commands in GB. Can you provide some pointers?